File: HW2/HW2.py
def main():
    holder = [2, 4, 3, 7, 9, 10]
    print(find_first(holder))

def find_first(holder):
    left = 0
    right = len(holder) - 1
    while left < right:
        m = (right - left + left) // 2
        if is_bad_version(holder[m]):
            right = m
        else:
            left = m + 1
    return holder[left]

def is_bad_version(version):
    return version % 2 == 1

if __name__ == "__main__":
    main()

File: HW2/HW2.java
package HW.HW2;


public class HW2 {
    
    public static void main(String[] args)
    {
        int [] holder = {2, 4, 3, 7, 9, 10};
        System.out.println(findFirst(holder));
    }

    public static int findFirst(int [] holder)
{
    int left = 0;
    int right = holder.length -1;
    while(left < right)
    {
        int m = ((right-left)+left)/2;
        if(isBadVersion(holder[m]))
        {
            right = m;
        }
        else
        {
            left = m+1;
        }
    }
    return holder[left];
}
public static boolean isBadVersion(int version)
{
    return version % 2 == 1;
}
}


